doc/nrf/protocols/matter/getting_started/memory_profiling.rst

```diff
@@ -0,0 +1,371 @@
+
+    CONFIG_SHELL=y
+    CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL=y
+
+The NVS Settings usage may change during the device's lifetime.
+In Matter samples, if you set the ``storage_partition``size, you cannot change it using a firmware update.
+It is possible only by reflashing the device with new code that includes a new partition layout.
+Therefore, it is crucial to calculate and set the appropriate ``storage_partition`` size according to the project requirements, such as enabled endpoints, the cryptographic backend used, or the number of supported Matter fabrics.
+
+If this functionality is enabled, you can use :ref:`ug_matter_configuring_settings_shell`.
+
+To see the full list of available commands, use the following UART shell command on your device:
+
+.. code-block:: console
+
+    uart:~$ matter_settings
+
+You will see a list of the available commands like this one:
+
+.. code-block:: console
+
+    peak      : Print peak settings size in Bytes. This size is reset during
+              reboot.
+              Usage: matter_settings peak
+    reset     : Reset peak settings size in Bytes.
+                Usage: matter_settings reset
+    get_size  : Get size of the chosen settings entry.
+                Usage: matter_settings get_size <name>
+    current   : Get current settings size in Bytes.
+                Usage: matter_settings current
+    free      : Get free settings space in Bytes.
+                Usage: matter_settings free
+
+Similarly to heap measurements, you can reset the current peak usage value, read the peak value, perform some operations on the device, and read the peak value again to obtain the difference.
+
+To reset the peak usage value, use the following UART shell command on your device:
+
+.. code-block:: console
+
+    uart:~$ matter_settings reset
+
+To measure the peak usage, use the following UART shell command on your device:
+
+.. code-block:: console
+
+    uart:~$ matter_settings peak
+
+You can also read a specific settings entry size by calling ``matter_settings get_size <name>`` UART shell command on your device.
+To obtain the name of an entry, you can use the ``settings list`` command form the ``settings`` UART shell command set.
+
+To read a specific settings entry size complete the following steps:
+
+1. Read all available settings list:
+
+    .. code-block:: console
+
+        uart:~$ settings list
+
+    .. code-block:: console
+
+        mt/g/im/ec
+        mt/g/gdc
+        mt/g/gcc
+        mt/g/lkgt
+        mt/ctr/reboot-count
+        mt/cfg/unique-id
+        its/0000000000020001
+
+#. Choose one of the available key, for example ``mt/ctr/reboot-count`` to read size of the reboot counter.
+#. Read the size of the chosen key:
+
+    .. code-block:: console
+
+        matter_settings get_size mt/ctr/reboot-count
+
+To learn about other ``settings`` UART shell commands use the following UART shell command on your device:
+
+.. code-block:: console
+
+    uart:~$ settings
```

Comment:
Same here, I do not get help after typing `settings` but error about missing subcomand.